src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,49 @@
+package duke;
+
+public class Parser {
+
+    /** Converts a task from a format of:
+     * TASK_TYPE | TASK_IS_DONE | TASK_NAME | TASK_START | TASK_END
+     * to a Task object
+     * @return a Task object
+     */
+    public static Task convertTaskFromLineInTaskList(String line) {
+        String[] taskInfo = line.split("[|]");
+        String taskType = taskInfo[0].trim();
+        boolean isDone = taskInfo[1].trim().equals("1");
+        String taskDescription = taskInfo[2].trim();
+        switch (taskType) {
+        case "T":
+            return new ToDos(taskDescription, isDone);
+        case "D":
+            String taskTime = taskInfo[3].trim();
+            return new Deadlines(taskDescription, taskTime, isDone);
+        case "E":
+            String startTime = taskInfo[3].trim();
+            String endTime = taskInfo[4].trim();
+            return new Events(taskDescription, startTime, endTime, isDone);

```

Comment:
Feel like this part can be improved through polymorphism

Reply:
Hmm, could you elaborate?